docs/dev_guide/compute.rst

```diff
@@ -0,0 +1,96 @@
+  ``X^rho`` etc, and derivatives by a single character subscript, ``X_r`` etc for :math:`\partial_{\rho} X`
+* ``label``: a LaTeX style label used for plotting.
+* ``units``: SI units of the quantity in LaTeX format
+* ``units_long``: SI units of the quantity, spelled out
+* ``description``: A short description of the quantity
+* ``dim``: Dimensionality of the quantity. Vectors (such as magnetic field) have ``dim=3``,
+  local scalar quantities (such as vector components, pressure, volume element, etc)
+  have ``dim=1``, global scalars (such as total volume, aspect ratio, etc) have ``dim=0``
+* ``params``: list of strings of ``Equilibrium`` parameters needed to compute the quantity
+  such as ``R_lmn``, ``Z_lmn`` etc. These will be passed into the compute function as a
+  dictionary in the ``params`` argument. Note that this only includes direct dependencies
+  (things that are used in this function). For most quantities, this will be an empty list.
+  For example, if the function relies on ``R_t``, this dependency should be specified as a
+  data dependency (see below), while the function to compute ``R_t`` itself will depend on
+  ``R_lmn``
+* ``transforms``: a dictionary of what ``transform`` objects are needed, with keys being the
+  quantity being transformed (``R``, ``p``, etc) and the values are a list of derivative
+  orders needed in ``rho``, ``theta``, ``zeta``. IE, if the quantity requires
+  :math:`R_{\rho}{\zeta}{\zeta}`, then ``transforms`` should be ``{"R": [[1, 0, 2]]}``
+  indicating a first derivative in ``rho`` and a second derivative in ``zeta``. Note that
+  this only includes direct dependencies (things that are used in this function). For most
+  quantities this will be an empty dictionary.
+* ``profiles``: List of string of ``Profile`` quantities needed, such as ``pressure`` or
+  ``iota``. Note that this only includes direct dependencies (things that are used in
+  this function). For most quantities this will be an empty list.
+* ``coordinates``: String denoting which coordinate the quantity depends on. Most will be
+  ``"rtz"`` indicating it is a function of :math:`\rho, \theta, \zeta`. Profiles and flux surface
+  quantities would have ``coordinates="r"`` indicating it only depends on `:math:\rho`
+* ``data``: What other physics quantities are needed to compute this quantity. In our
+  example, we need the radial derivative of pressure ``p_r``, the Jacobian determinant
+  ``sqrt(g)``, and contravariant components of current and magnetic field. These dependencies
+  will be passed to the compute function as a dictionary in the ``data`` argument. Note
+  that this only includes direct dependencies (things that are used in this function).
+  For example, we need ``sqrt(g)``, which itself depends on ``e_rho``, etc. But we don't
+  need to specify ``e_rho`` here, that dependency is determined automatically at runtime.
+* ``kwargs``: If the compute function requires any additional arguments they should
```

Comment:
add link to a cmpute fxn that uses this, or change it to be `f_C` so `helicity` kwarg is shown #1332